src/conversion/azure/common.ts

```diff
@@ -0,0 +1,44 @@
+import {NodeSpec} from '../../graph';
+
+import {IEntityStore} from '..';
+
+import {AnyAzureObject} from '.';
+
+export interface ItemMoniker {
+  readonly item: AnyAzureObject;
+  readonly name: string;
+}
+
+// DESIGN NOTE: Both monikers and convert return an array or list
+// of items as it is possible that a top level item in the Azure
+// graph consists of internal elements which may be referenced by
+// other items in teh graph. An example of this could be network
+// interface card on a Virutal Machine in a scale set
+export interface IAzureConverter<T extends AnyAzureObject> {
+  readonly supportedType: string;
+  monikers(input: T): ItemMoniker[];
+  convert(input: T, stores: IEntityStore<AnyAzureObject>): NodeSpec[];
+}
+
+export function extractMonikers(item: AnyAzureObject): ItemMoniker[] {
+  return [{item, name: item.name}];
+}
+
+// DESIGN NOTE: Skip processsing NodeSpecs exists for items which
+// need to be indexed as their data is used by other elements
+// however these types do not make sense to be converted indpendently
+function skipProcessingNodeSpecs(
```

Comment:
I am still a bit skeptical of this idea that all converters are related by an abstract base class.

This decision results in lines 27-44, devoted solely to AzureObjects that don't have converters. The premise here is that a default converter could somehow throw and exception or log unconverted types. Throwing or logging is really the only option, in this scenario, because the return value of convert() is NodeSpec[].

Reply:
As we discussed in our code review. It feels like there another change that will likely emerge. I'd like to get this PR in to enable some of your other work, then evolve this code as we start to see a more complete Azure Graph